diff --git a/src/java/com/android/internal/telephony/RIL.java b/src/java/com/android/internal/telephony/RIL.java
index 41014635..1af847cb 100644
--- a/src/java/com/android/internal/telephony/RIL.java
+++ b/src/java/com/android/internal/telephony/RIL.java
@@ -327,7 +327,13 @@ public class RIL extends BaseCommands implements CommandsInterface {
     static final int RESPONSE_SOLICITED_ACK_EXP = 3;
     static final int RESPONSE_UNSOLICITED_ACK_EXP = 4;
 
-    static final String[] SOCKET_NAME_RIL = {"rild", "rild2", "rild3"};
+    static final String[] SOCKET_NAME_RIL;
+    static {
+        String rilSocketName = SystemProperties.get("ro.telephony.ril.socket_name");
+        SOCKET_NAME_RIL = rilSocketName.isEmpty()
+                ? new String[] { "rild", "rild2", "rild3" }
+                : rilSocketName.split(",");
+    }
 
     static final int SOCKET_OPEN_RETRY_MILLIS = 4 * 1000;
 
